include/ghex/communication_object_ipr.hpp

```diff
@@ -0,0 +1,564 @@
+/* 
+ * GridTools
+ * 
+ * Copyright (c) 2014-2020, ETH Zurich
+ * All rights reserved.
+ * 
+ * Please, refer to the LICENSE file in the root directory.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * 
+ */
+#ifndef INCLUDED_GHEX_COMMUNICATION_OBJECT_IPR_HPP
+#define INCLUDED_GHEX_COMMUNICATION_OBJECT_IPR_HPP
+
+// TO DO: remove unneeded includes
+#include "./cuda_utils/stream.hpp"
+#include "./packer.hpp"
+#include "./common/utils.hpp"
+#include "./common/test_eq.hpp"
+#include "./buffer_info.hpp"
+#include "./transport_layer/tags.hpp"
+#include "./arch_traits.hpp"
+#include <map>
+#include <unordered_map>
+#include <queue>
+#include <stdio.h>
+#include <functional>
+
+namespace gridtools {
+
+    namespace ghex {
+
+        // namespace unstructured {
```

Comment:
remove? Ho do we protect against mis-use?

Reply:
Done, used a specialization for unstructured.